src/EntityGraphQL/Compiler/GqlNodes/ExecutableGraphQLStatement.cs

```diff
@@ -100,18 +100,18 @@ public ExecutableGraphQLStatement(ISchemaProvider schema, string name, Expressio
                     var errors = aex.InnerExceptions.SelectMany<Exception, string>(ex => ex is EntityGraphQLValidationException vex ? vex.ValidationErrors : new[] { $"Field '{fieldNode.Name}' - {ex.Message}" });
                     throw new EntityGraphQLValidationException(errors);
                 }
-                catch (TargetInvocationException ex)
-                {
-                    if (ex.InnerException is EntityGraphQLException vex)
-                        throw new EntityGraphQLException(fieldNode.Name, vex);
-                    throw new EntityGraphQLExecutionException(fieldNode.Name, ex.InnerException!);
-                }
                 catch (EntityGraphQLValidationException)
                 {
                     throw;
                 }
                 catch (Exception ex)
                 {
+                    while (ex is TargetInvocationException)
```

Comment:
does GetBaseException not work in this case?
https://docs.microsoft.com/en-us/dotnet/api/system.exception.getbaseexception?view=net-6.0

Reply:
It depends where in the exception chain is most valuable. Sometimes the base exception is too narrow. For example, suppose a service is fetching from an API and the network connection fails. The base exception may be a low-level SocketException. It may be wrapped in a HttpException, which itself may be wrapped by the service layer in a MyServiceException. All of this may then be wrapped in two layers of TargetInvocationException by EntityGraphQL. In this example, the MyServiceException would likely be the most valuable.

Since EntityGraphQL has the luxury of JSON for the output, perhaps it can provide an array or linked list of all the exception layers (except the non-useful ones like TargetInvocationException).

Reply:
This fits the spec as well. A result (with errors) is actually a list of errors.

`QueryResult.AddError()` can be used. Currently there are a few places trying to catch exceptions and clean them up. Might be nice to bring that to one if we can.

`ExecutableGraphQLStatement` and `GraphQLMutationStatement` mainly catch exceptions to add `field.Name` to the detail.

In `SchemaProvider.ExecuteRequestAsync` it does some "unwrapping" should the logic in this PR be in that location to do the same for anything executed?

Reply:
I found the location for the PR based based on the inspecting in the debugger the stack trace of a first-chance exception my code threw. I didn't look holistically to see where else unwrapping logic would be helpful. Factoring out unwrapping (or even better cataloging) exception layers into a common helper method makes sense to me.